### PR TITLE
Add FIPS_VARIATION_PROBLEM_LIST_FILE support

### PR DESCRIPTION
**Pull request by llxia**
related: backlog/issues/1529 and https://github.com/adoptium/aqa-tests/pull/5662

### CODE REVIEW

**Review by LongyuZhang — APPROVED** [review]
LGTM

File: src/org/testKitGen/MkGen.java [hunk 1]
```diff
@@ -96,7 +96,21 @@ private void writeSingleTest(List<String> testsInPlaylist, TestInfo testInfo, Fi
 				f.write(testTargetName + ": TEST_GROUP=" + testInfo.getLevelStr() + "\n");
 				f.write(testTargetName + ": TEST_ITERATIONS=" + testInfo.getIterations() + "\n");
 				f.write(testTargetName + ": AOT_ITERATIONS=" + testInfo.getAotIterations() + "\n");
-
+				
+				// Set special openjdk problem list for JVM options that contains FIPS profile
+				// This feature is ignored if TEST_FLAG contains FIPS
+				if (arg.getBuildList().contains("openjdk")) {
+					if (!arg.getTestFlag().contains("FIPS") && !var.getJvmOptions().isEmpty() && var.getJvmOptions().contains("-Dsemeru.customprofile=")) {
```
Comment (JasonFengJ9) [inline src/org/testKitGen/MkGen.java:103]: `var.getJvmOptions()` can be cached in a local variable.

File: src/org/testKitGen/MkGen.java [hunk 2]
```diff
@@ -96,7 +96,21 @@ private void writeSingleTest(List<String> testsInPlaylist, TestInfo testInfo, Fi
...
+						String[] splited = var.getJvmOptions().split("\\s+");
+						for (int i=0; i < splited.length; i++) {
```
Comment (JasonFengJ9) [inline src/org/testKitGen/MkGen.java:105]:
```suggestion
						for (int i = 0; i < splited.length; i++) {
```

File: src/org/testKitGen/MkGen.java (hunk 1 above)
Comment (JasonFengJ9) [reply src/org/testKitGen/MkGen.java:103]: `-Dsemeru.customprofile=` can be assigned to a constant string for later multiple usages.

File: src/org/testKitGen/MkGen.java [hunk 3]
```diff
@@ -96,7 +96,21 @@ private void writeSingleTest(List<String> testsInPlaylist, TestInfo testInfo, Fi
...
+							if (splited[i].contains("-Dsemeru.customprofile=")) {
+								String fipsProfile = splited[i].replace("-Dsemeru.customprofile=", "").trim();
+								f.write(testTargetName + ": FIPS_VARIATION_PROBLEM_LIST_FILE=-exclude:$(Q)$(JTREG_JDK_TEST_DIR)/ProblemList-" + fipsProfile + ".txt$(Q)\n");
```
Comment (JasonFengJ9) [inline src/org/testKitGen/MkGen.java:108]: Should this `break`, or continue searching for other `splited[i]`?

Comment (llxia) [reply src/org/testKitGen/MkGen.java:108]: We should break. Ideally, we only have one customprofile.

File: src/org/testKitGen/MkGen.java (hunk 1 above)
Comment (llxia) [reply src/org/testKitGen/MkGen.java]: updated

**Review by JasonFengJ9 — APPROVED** [review]

**Pull request merged**